packages/api/src/beacon/routes/beacon/index.ts

```diff
@@ -28,51 +26,41 @@ export type {
   StateId,
   ValidatorId,
   ValidatorStatus,
-  ValidatorFilters,
-  CommitteesFilters,
   FinalityCheckpoints,
   ValidatorResponse,
   ValidatorBalance,
   EpochCommitteeResponse,
   EpochSyncCommitteeResponse,
 } from "./state.js";
 
-export type Api = block.Api &
-  pool.Api &
-  state.Api &
-  rewards.Api & {
-    getGenesis(): Promise<ApiClientResponse<{[HttpStatusCode.OK]: {data: phase0.Genesis}}>>;
+export type Endpoints = block.Endpoints &
+  pool.Endpoints &
+  state.Endpoints &
+  rewards.Endpoints & {
+    getGenesis: Endpoint<
+      //
+      "GET",
+      EmptyArgs,
+      EmptyRequest,
+      phase0.Genesis,
+      EmptyMeta
+    >;
   };
 
-export const routesData: RoutesData<Api> = {
-  getGenesis: {url: "/eth/v1/beacon/genesis", method: "GET"},
-  ...block.routesData,
-  ...pool.routesData,
-  ...state.routesData,
-  ...rewards.routesData,
-};
-
-export type ReqTypes = {
-  [K in keyof ReturnType<typeof getReqSerializers>]: ReturnType<ReturnType<typeof getReqSerializers>[K]["writeReq"]>;
-};
-
-// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
-export function getReqSerializers(config: ChainForkConfig) {
-  return {
-    getGenesis: reqEmpty,
-    ...block.getReqSerializers(config),
-    ...pool.getReqSerializers(),
-    ...state.getReqSerializers(),
-    ...rewards.getReqSerializers(),
-  };
-}
-
-export function getReturnTypes(): ReturnTypes<Api> {
+export function definitions(config: ChainForkConfig): RouteDefinitions<Endpoints> {
   return {
-    getGenesis: ContainerData(ssz.phase0.Genesis),
-    ...block.getReturnTypes(),
-    ...pool.getReturnTypes(),
-    ...state.getReturnTypes(),
-    ...rewards.getReturnTypes(),
+    getGenesis: {
+      url: "/eth/v1/beacon/genesis",
+      method: "GET",
+      req: EmptyRequestCodec,
+      resp: {
+        data: ssz.phase0.Genesis,
+        meta: EmptyMetaCodec,
+      },
+    },
+    ...block.definitions(config),
+    ...pool.definitions,
+    ...state.definitions,
+    ...rewards.definitions,
```

Comment:
Make it consistent pattern that all `routeDefinitions` should accept same parameters.

Reply:
Good point, I looked at this before but noted it adds an extra indentation due to extra return nesting but I have updated this now to make it more consistent. Also avoids potential work in the future if new apis start requiring `config`.